### PR TITLE
Redesign UI. Implement Picker View

### PR DESCRIPTION
**Pull request by marcusykim**
This was initially a branch that was created to build the beat card selection feature. Many other changes were made along the way. I realized that the commits seem random, but there was a lot of trial and error here. The update from AudioKit 4 to AudioKit 5 was also nontrivial and require a lot of rework. The McPicker library was added successfully and allows users to select a beat card from a picker view. This selection feature still needs to be completed, as the audio does not follow any changes that the picker makes to the UI.

### CODE REVIEW

File: Four Brains/Model/DrumSounds.swift [hunk 1]
```diff
@@ -67,9 +80,9 @@ class DrumSounds {
         for beatCardCounter in Range(0...3) {
             for note in Range(0...3) {
                 if beatCardCounter == 0 && note == 0{
-                    rideNoteSequence[0] = wholeBeat.ridePattern[beatCardCounter].beatCardNoteSequence[note]
+                    rideNoteSequence[0] = wholeBeat.getRidePattern()[beatCardCounter].getBeatCardNoteSequence()[note]
```
Comment (bradenaa) [inline Four Brains/Model/DrumSounds.swift]: How come we are explicitly setting the 0 index like this?

File: Four Brains/Model/DrumSounds.swift [hunk 2]
```diff
@@ -79,9 +92,9 @@ class DrumSounds {
         for beatCardCounter in Range(0...3) {
             for note in Range(0...3) {
                 if beatCardCounter == 0 && note == 0{
-                    snareNoteSequence[0] = wholeBeat.snarePattern[beatCardCounter].beatCardNoteSequence[note]
+                    snareNoteSequence[0] = wholeBeat.getSnarePattern()[beatCardCounter].getBeatCardNoteSequence()[note]
```
Comment (bradenaa) [inline Four Brains/Model/DrumSounds.swift]: same here

File: Four Brains/Model/DrumSounds.swift [hunk 3]
```diff
@@ -103,29 +116,31 @@ class DrumSounds {
...
+        if !mute.getRideMuteState(){
             for note in Range(0...15) {
                 let position = ((Double(note) + 1.0)/4.0) - 0.25
                 if rideNoteSequence[note] == 1 {
-                    sequencer.tracks[0].add(noteNumber: 34, velocity: 200, position: AKDuration(beats: position), duration: AKDuration(beats: 1.0))
+                    sequencer.tracks[0].add(
+                        noteNumber: 34,
+                        velocity: 200,
+                        position: Duration(beats: position),
+                        duration: Duration(beats: 0.25))
```
Comment (bradenaa) [inline Four Brains/Model/DrumSounds.swift:172]: This is a lot better from a readability perspective. Is there such thing as a linter in swift?

File: Four Brains/Model/DrumSounds.swift [hunk 4]
```diff
@@ -177,29 +204,29 @@ class DrumSounds {
...
-            let session = AVAudioSession.sharedInstance()
-            try session.setCategory(AVAudioSession.Category.playAndRecord)
-
-            if !AKSettings.headPhonesPlugged {
-                try session.overrideOutputAudioPort(AVAudioSession.PortOverride.speaker)
-         }
-        }catch {
-            print("error in settings.setSession")
-        }
+//        do {
```
Comment (bradenaa) [inline Four Brains/Model/DrumSounds.swift:237]: How come this is commented out?

File: Four Brains/Model/K.swift [hunk 5]
```diff
@@ -9,24 +9,110 @@
 import Foundation
 
 class K {
-    class MUTECONSTANTS {
+    class MuteConstants {
```
Comment (bradenaa) [inline Four Brains/Model/K.swift]: Nice. Not sure on swift standards for naming variables, but this is easier for me to read

File: Four Brains/Model/K.swift [hunk 6]
```diff
@@ -9,24 +9,110 @@
...
+        static let BEAT_CARD_2B_NOTE_STRING = "- X X -"
+        static let BEAT_CARD_2C_NOTE_STRING = "- - X X"
+        static let BEAT_CARD_2D_NOTE_STRING = "X - - X"
+        static let BEAT_CARD_2E_NOTE_STRING = "X - X -"
+        static let BEAT_CARD_2F_NOTE_STRING = "- X - X"
+        static let BEAT_CARD_3A_NOTE_STRING = "X X X -"
+        static let BEAT_CARD_3B_NOTE_STRING = "X X - X"
+        static let BEAT_CARD_3C_NOTE_STRING = "X - X X"
+        static let BEAT_CARD_3D_NOTE_STRING = "- X X X"
+        static let BEAT_CARD_4_NOTE_STRING = "X X X X"
```
Comment (bradenaa) [inline Four Brains/Model/K.swift]: Will these variables ever be used? Where the X is, is very reminiscent of where the one is in the note sequence array above. Could these strings just be generated via the above definitions?

File: Four Brains/Model/K.swift [hunk 7]
```diff
@@ -9,24 +9,110 @@
...
+        static let BEAT_CARD_2B = BeatCard(label: "2B", noteSequence: [0,1,1,0])
+        static let BEAT_CARD_2C = BeatCard(label: "2C", noteSequence: [0,0,1,1])
+        static let BEAT_CARD_2D = BeatCard(label: "2D", noteSequence: [1,0,0,1])
+        static let BEAT_CARD_2E = BeatCard(label: "2E", noteSequence: [1,0,1,0])
+        static let BEAT_CARD_2F = BeatCard(label: "2F", noteSequence: [0,1,0,1])
+        static let BEAT_CARD_3A = BeatCard(label: "3A", noteSequence: [1,1,1,0])
+        static let BEAT_CARD_3B = BeatCard(label: "3B", noteSequence: [1,1,0,1])
+        static let BEAT_CARD_3C = BeatCard(label: "3C", noteSequence: [1,0,1,1])
+        static let BEAT_CARD_3D = BeatCard(label: "3D", noteSequence: [0,1,1,1])
+        static let BEAT_CARD_4 = BeatCard(label: "4", noteSequence: [1,1,1,1])
```
Comment (bradenaa) [inline Four Brains/Model/K.swift]: Will these ever change? Looks very manual

**Review by bradenaa — COMMENTED** [review]
Got half way through. Left a couple comments. Will finish and look through the rest later

**Comment by bradenaa** [comment]
Can you leave a detailed summary above?

https://cbea.ms/git-commit/

It will be easier to understand what changes you have made if there is a human readable summary to first parse through.

**Comment by bradenaa** [comment]
https://cbea.ms/git-commit/

File: Four Brains/Model/DrumSounds.swift (hunk 4 above)
Comment (bradenaa) [reply Four Brains/Model/DrumSounds.swift:237]: Maybe leave a TODO: ... get back to this later, not working right now...

File: Four Brains/Model/K.swift (hunk 7 above)
Comment (bradenaa) [reply Four Brains/Model/K.swift]: Ok seems like this won't change, but maybe make a note in the code about that

File: Four Brains/Model/DrumSounds.swift (hunk 1 above)
Comment (marcusykim) [reply Four Brains/Model/DrumSounds.swift]: Thanks, I've fixed this. It was unnecessarily overwriting the `0` that I initialized the array with. Updated the initialization to an empty array. No need for the unnecessary `if` statement above.

File: Four Brains/Model/DrumSounds.swift (hunk 2 above)
Comment (marcusykim) [reply Four Brains/Model/DrumSounds.swift]: *see above comment

File: Four Brains/Model/DrumSounds.swift (hunk 3 above)
Comment (marcusykim) [reply Four Brains/Model/DrumSounds.swift:172]: Thanks!

Comment (marcusykim) [reply Four Brains/Model/DrumSounds.swift:172]: After this comment, I installed a linter called swiftlint. It's working well.

File: Four Brains/Model/DrumSounds.swift (hunk 4 above)
Comment (marcusykim) [reply Four Brains/Model/DrumSounds.swift:237]: I went ahead and left the TODO. Thanks!

File: Four Brains/Model/K.swift (hunk 5 above)
Comment (marcusykim) [reply Four Brains/Model/K.swift]: Thanks! Camel case is the standard for this usage

File: Four Brains/Model/K.swift (hunk 6 above)
Comment (marcusykim) [reply Four Brains/Model/K.swift]: I will create a new branch after this merge and refactor this to generate the beat cards. Thanks!

**Pull request merged**